### DISCUSSION TITLE
get rid of phase and set all pixels in grain to mean grain orientation

### DISCUSSION THREAD

**Discussion by miamaric12** [mtex-toolbox/mtex#2244] [Ask Anything]
Hello, 

I am wondering if this is possible in mtex. I am trying to get rid of my second phase (hydride) and assign the points in the hydride and neighbouring points within the grain (that have a lot of misorientation) the mean grain orientation. I have tried to do this by removing the phase and filling in data. It works mostly, but could you help improve it? 

the big issue i am having is exporting the new ctf, when i try and read it in it does not look right (as shown in the figures) 

Here is my code: 
```matlab
%% Specify Crystal and Specimen Symmetries

% crystal symmetry
CS = {... 
  'notIndexed',...
  crystalSymmetry('m-3m', [4.777 4.777 4.777], 'mineral', 'ZircHydride', 'color', [0.53 0.81 0.98]),...
  crystalSymmetry('6/mmm', [3.232 3.232 5.147], 'X||a*', 'Y||b', 'Z||c*', 'mineral', 'Zirc-alloy4', 'color', [0.56 0.74 0.56])};

% plotting convention
setMTEXpref('xAxisDirection','east');
setMTEXpref('zAxisDirection','IntoPlane');

%% Specify File Names

% path to files
pname = 'C:\Users\Rhys\Desktop';
fname = [pname '\LBwT_ 26.ctf'];

%% Import the Data

% create an EBSD variable containing the data
ebsd = EBSD.load(fname,CS,'interface','ctf',...
  'convertEuler2SpatialReferenceFrame');

ebsd = ebsd.gridify;

ipfKey = ipfHSVKey(ebsd('Zirc-Alloy4').CS.properGroup);
color = ipfKey.orientation2color(ebsd('Zirc-Alloy4').orientations);
%plot(ebsd('Zirc-Alloy4'), color)


[grains,ebsd.grainId] = calcGrains(ebsd('indexed'), 'angle', 5*degree)


% consider only indexed data
ebsd = ebsd('Zirc-Alloy4');

% reconstruct the grain structure
[grains,ebsd.grainId,ebsd.mis2mean] = calcGrains(ebsd,'angle',5*degree);


% interpolate the missing data
ebsd_filled = fill(ebsd,grains);

ebsd = ebsd_filled('indexed');

[grains,ebsd.grainId,ebsd.mis2mean] = calcGrains(ebsd,'angle',5*degree);


% define the meanFilter
F = meanFilter;
F.numNeighbours = 10;

% smooth the data
ebsdS = smooth(ebsd,F);
ebsdS = ebsdS('indexed');

plot(ebsd('Zirc-Alloy4'),ebsd('Zirc-Alloy4').orientations,'coordinates','on')

ebsdS.export('myFile.ctf')
```

Thank you for the help. 
Before export: 
![image (2)](https://github.com/user-attachments/assets/315d2612-68f6-4152-9804-fb7eb538f183)
![image (1)](https://github.com/user-attachments/assets/b0adc8f4-2a74-4210-b32b-c5b792f33cf4)
After export:
![image](https://github.com/user-attachments/assets/b4603821-bd15-4f28-a740-36a38ad51618)

**Comment by ralfHielscher** [comment]
Does it look correct within MTEX?

**Reply by miamaric12** [reply]
Yes when I plot it in mtex before exporting it is fine but after it is not

**Reply by ralfHielscher** [reply]
Can MTEX read the exported ctf file?
